src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.DevOps.tests.ps1

```diff
@@ -21,12 +21,50 @@ InModuleScope Arcus.Scripting.DevOps {
                     # Assert
                     $getResponse = Invoke-WebRequest -Uri $requestUri -Method Get -Headers $headers
                     $json = ConvertFrom-Json $getResponse.Content
-                    $json.keepForever | Should -Be $true
+                    foreach ($lease in $json.value) {
+                        $lease.protectPipeline | Should -Be $true
+                        $date = Get-Date -Year 2200 -Month 1 -Day 1
+                        $lease.validUntil | Should -BeGreaterThan $date
+                    }
                 } finally {
-                    $retentionPayload = @{ keepforever='false' }
-                    $requestBody = $retentionPayload | ConvertTo-Json -Compress
-                    $patchResponse = Invoke-WebRequest -Uri $requestUri -Method Patch -Headers $headers -Body $requestBody -ContentType "application/json"
-                    $patchResponse.StatusCode | Should -Be 200
+                    $getResponse = Invoke-WebRequest -Uri $requestUri -Method Get -Headers $headers
+                    $json = ConvertFrom-Json $getResponse.Content
+                    foreach ($lease in $json.value) {
+                        $deleteUri = "$collectionUri" + "$projectId/_apis/build/retention/leases?ids=" + $lease.leaseId + "&api-version=7.0"
+                        $deleteResponse = Invoke-WebRequest -Uri $deleteUri -Method Delete -Headers $headers
+                        $deleteResponse.StatusCode | Should -Be 204
+                    }
+                }
+            }
+            It "Saves the Azure DevOps build for 10 days" {
+                # Arrange
+                $projectId = $env:SYSTEM_TEAMPROJECTID
+                $buildId = $env:BUILD_BUILDID
+                $collectionUri = $env:SYSTEM_COLLECTIONURI
+                $requestUri = "$collectionUri" + "$projectId/_apis/build/builds/" + $buildId + "/leases?api-version=7.0"
+                $headers = @{ Authorization = "Bearer $env:SYSTEM_ACCESSTOKEN" }
+                try {
+                    # Act
+                    Save-AzDevOpsBuild -ProjectId $projectId -BuildId $buildId -DaysToKeep 10
+
+                    # Assert
+                    $getResponse = Invoke-WebRequest -Uri $requestUri -Method Get -Headers $headers
+                    $json = ConvertFrom-Json $getResponse.Content
+                    foreach ($lease in $json.value) {
+                        $lease.protectPipeline | Should -Be $true
+                        $dateGreaterThan = (Get-Date).AddDays(9)
+                        $dateLessThan = (Get-Date).AddDays(11)
+                        $lease.validUntil | Should -BeGreaterThan $dateGreaterThan
+                        $lease.validUntil | Should -BeLessThan $dateLessThan
```

Comment:
Good idea to assert with these threshold, a real example of 'flexible assertions'. 👍